core/src/main/scala/org/apache/gearpump/cluster/AppDescription.scala

```diff
@@ -139,4 +138,47 @@ case class ExecutorContext(
 case class ExecutorJVMConfig(
     classPath: Array[String], jvmArguments: Array[String], mainClass: String,
     arguments: Array[String], jar: Option[AppJar], username: String,
-    executorAkkaConfig: Config = ConfigFactory.empty())
\ No newline at end of file
+    executorAkkaConfig: Config = ConfigFactory.empty())
+
+// Not using Enumeration here because upickle does not support it
+sealed trait ApplicationStatus {
+  def isTerminalStatus: Boolean
+}
+
+object ApplicationStatus {
+  case object Pending extends ApplicationStatus {
+    override def toString: String = "pending"
```

Comment:
You may define this only once in the superclass and pass in the status string from subclasses.